spring-security-modules/spring-security-mvc-boot-2/src/main/java/com/baeldung/loginredirect/LoginPageInterceptor.java

```diff
@@ -0,0 +1,39 @@
+package com.baeldung.loginredirect;
+
+import org.apache.http.HttpStatus;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.util.UrlPathHelper;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+class LoginPageInterceptor extends HandlerInterceptorAdapter {
```

Comment:
do we want to make this package protected for some reason? if not let's make it public (check the others classes as well)

Reply:
It is recommended in the _**baeldung - for authors - 2 - code in the articles**_ document, to use the package-scope access modifier if possible.